deps/thpool/thpool.c

```diff
@@ -1058,23 +1055,19 @@ void redisearch_thpool_schedule_config_reduce_threads_job(redisearch_thpool_t *t
   if (thpool_p->state == THPOOL_UNINITIALIZED || thpool_p->n_threads == 0) {
     if (thpool_p->n_threads > 0) {
       // If is UNINITIALIZED, at least it would lazily initialize less threads
-      assert(thpool_p->n_threads >= n_threads_to_remove && "Number of threads can't be negative");
+      RS_ASSERT(thpool_p->n_threads >= n_threads_to_remove && "Number of threads can't be negative");
       thpool_p->n_threads -= n_threads_to_remove;
     }
     return;
   }
 
   size_t n_threads = thpool_p->n_threads;
   LOG_IF_EXISTS("verbose", "Scheduling from main thread to remove %zu threads", n_threads_to_remove);
-  assert((!remove_all || n_threads_to_remove == n_threads) && "If remove_all is set, n_threads_to_remove must be equal to n_threads");
-  assert(thpool_p->n_threads >= n_threads_to_remove && "Number of threads can't be negative");
-  assert(thpool_p->jobqueues.state == JOBQ_RUNNING && "Can't remove threads while jobq is paused");
-  assert(thpool_p->num_threads_alive >= n_threads_to_remove && "Not enough alive threads to remove");
-
-  ThreadState new_state = remove_all ? THREAD_TERMINATE_WHEN_EMPTY : THREAD_TERMINATE_ASAP;
+  RS_LOG_ASSERT((!terminate_when_empty || n_threads_to_remove == n_threads), "If remove_all is set, n_threads_to_remove must be equal to n_threads");
+  RS_ASSERT(thpool_p->n_threads >= n_threads_to_remove && "Number of threads can't be negative");
+  RS_ASSERT(thpool_p->jobqueues.state == JOBQ_RUNNING && "Can't remove threads while jobq is paused");
```

Comment:
can we do it for these as well?
```suggestion
  RS_LOG_ASSERT(thpool_p->n_threads >= n_threads_to_remove, "Number of threads can't be negative");
  RS_LOG_ASSERT(thpool_p->jobqueues.state == JOBQ_RUNNING, "Can't remove threads while jobq is paused");
```